.github/workflows/build_wheels.yml

```diff
@@ -101,11 +112,20 @@ jobs:
           name: sdist_files
           path: dist
 
-      - name: Publish package to TestPyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          password: ${{ secrets.TEST_PYPI_TOKEN  }}
-          repository-url: https://test.pypi.org/legacy/
+
+      # The step below can be uncommented for testing / debugging purposes.
+
+      # We do not publish to TestPyPI here, as the job will fail if the version
+      # number already exists on TestPyPI.
+
+      # Note: if the non-release builds may fail to upload to TestPyPI if the version
+      # number is "non-local": https://github.com/pypa/setuptools_scm/issues/342
+
+      # - name: Publish package to TestPyPI
```

Comment:
think this is fine. I might need a mental note to check if we can parameterize the github actions run so that one can pass a "test" parameter to the run which will then either publish to TestPyPI or just skip the last step. 

This is what I found with a quick search:
 - https://stackoverflow.com/a/62736541
 - https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

Reply:
With testing I mean uncommenting the release step and just running the pipeline up to that point.

Reply:
From a different thread I see:

> usage of local versions is intended to prevent accident release uploads

which makes sense. From https://github.com/pypa/setuptools_scm/issues/342